src/main/java/com/timgroup/statsd/StatsDProcessor.java

```diff
@@ -15,12 +15,11 @@
 import java.util.concurrent.BlockingQueue;
 import java.util.concurrent.ConcurrentLinkedQueue;
 import java.util.concurrent.CountDownLatch;
-import java.util.concurrent.ExecutorService;
-import java.util.concurrent.Executors;
 import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.TimeUnit;
 import java.util.concurrent.atomic.AtomicInteger;
 
-public abstract class StatsDProcessor implements Runnable {
+public abstract class StatsDProcessor {
```

Comment:
No - previously the only reason this was a `Runnable` was because a separate thread-pool was used to start the worker threads (which had their own thread-pool.) Now that the intermediate thread-pool has been removed this class no longer needs to implement `Runnable`.